internal/report/report_test.go

```diff
@@ -0,0 +1,852 @@
+// Copyright 2023 Adevinta
+
+package report
+
+import (
+	"fmt"
+	"testing"
+
+	vreport "github.com/adevinta/vulcan-report"
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+
+	"github.com/adevinta/lava/internal/config"
+	"github.com/adevinta/lava/internal/engine"
+)
+
+func TestCalculateErrCode(t *testing.T) {
```

Comment:
I would rename it to:

```suggestion
func TestWriter_calculateExitCode(t *testing.T) {
```

After applying the other suggestions.